src/test/java/teammates/test/pageobjects/InstructorFeedbackEditPage.java

```
@@ -529,43 +525,104 @@ public boolean verifyNewRankOptionsQuestionFormIsDisplayed() {
+
+            for (WebElement date : dates) {
+
+                boolean isDisabled = date.getAttribute("class").contains("ui-datepicker-unselectable ui-state-disabled");
+                boolean isFromOtherMonth = date.getAttribute("class").contains("ui-datepicker-other-month");
+
+                if (isDisabled && !isFromOtherMonth) {
+                    return false;
+                }
             }
 
-            // different months with the same day (e.g. first week or last week of the month) in the datepicker
-            List<WebElement> daysWithSameDayAsYesterday = browser.driver.findElements(By.xpath("//td[a[text() = \"" + day + "\"]]"));
-            
-            for (WebElement yesterday : daysWithSameDayAsYesterday) {
-                // yesterday is enabled
-                if (!yesterday.getAttribute("class").contains("ui-datepicker-unselectable ui-state-disabled")) {
+            // Navigate to the next month
+            browser.driver.findElement(By.className("ui-datepicker-next")).click();
+        }
+
+        return true;
+    }
+
+    /**
+     * Navigate the datepicker associated with {@code dateBox} to the specified {@code date}
+     * 
+     * @param dateBox
+     * @param date
+     * @return true if navigated to the {@code date} successfully, otherwise
+     *         false
+     */
+    private boolean navigate(WebElement dateBox, Calendar date) {
+
+        dateBox.click();
+
+        Calendar selectedDate = Calendar.getInstance();
+
+        String month = date.getDisplayName(Calendar.MONTH, Calendar.LONG, Locale.ENGLISH);
+        String year = Integer.toString(date.get(Calendar.YEAR));
+
+        SimpleDateFormat dateFormat = new SimpleDateFormat("dd/MM/yyyy");
+
+        try {
+            selectedDate.setTime(dateFormat.parse(dateBox.getAttribute("value")));
+        } catch (ParseException e) {
+            e.printStackTrace();
```

The test can't continue if the parse fails, so let the test fail here instead of silencing it.